lib/DxcSupport/HLSLOptions.cpp

```diff
@@ -144,6 +144,30 @@ bool DxcOpts::IsLibraryProfile() {
   return TargetProfile.startswith("lib_");
 }
 
+bool DxcOpts::IsDebugInfoEnabled() {
+  return DebugInfo || !DebugFile.empty();
+}
+
+bool DxcOpts::EmbedDebugInfo() {
+  return DebugInfo && DebugFile.empty() && !StripDebug;
+}
+
+bool DxcOpts::EmbedPDBName() {
+  return IsDebugInfoEnabled() || !DebugFile.empty() || !PDBAltPath.empty();
+}
+
+bool DxcOpts::DebugFileIsDirectory() {
+  return !DebugFile.empty() && DebugFile.endswith(llvm::StringRef("\\"));
```

Comment:
I'd also test for `/` since Windows accepts it. It'd be unexpected to pass in `c:/foo/` and have this not kick in.

Reply:
Good point, I’ll add that.

Reply:
`llvm::sys::path::is_separator` does what you want.